### PR TITLE
feat(Util): allow array for StringOptions' char

### PR DESCRIPTION
**Pull request by u9g**
**Please describe the changes this PR makes and why it should be merged:**

Allow StringOptions to accept a string array for char

- https://github.com/discordjs/discord.js/issues/2721

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: src/util/Util.js [hunk 1]
```diff
@@ -64,6 +64,7 @@ class Util {
   static splitMessage(text, { maxLength = 2000, char = '\n', prepend = '', append = '' } = {}) {
     text = Util.resolveString(text);
     if (text.length <= maxLength) return [text];
+    if (Array.isArray(char)) char = char.filter(splitChar => text.includes(splitChar))[0];
```
Comment (VoltrexMaster) [inline src/util/Util.js:67]:
```suggestion
    if (Array.isArray(char)) char = char.find(splitChar => text.includes(splitChar));
```
No need to use `Array.prototype.filter()` just to get the first result.

Comment (u9g) [reply src/util/Util.js:67]: Fixed, thanks

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Comment by SpaceEEC** [comment]
I'm not sure if this approach (only using the first provided char that is in the content of the message) is good, I'd prefer something described in the linked issue that actually utilizes (if necessary) all chars in the array.
This also needs to be documented.

**Comment by kyranet** [comment]
@u9g any update on this? You have a pending request from Space.

**Comment by u9g** [comment]
> I'm not sure if this approach (only using the first provided char that is in the content of the message) is good, I'd prefer something described in the linked issue that actually utilizes (if necessary) all chars in the array.
> This also needs to be documented.

Documented and added support for arrays and regex expressions


> @u9g any update on this? You have a pending request from Space.

Just committed :)

**Review by SpaceEEC — DISMISSED** [review]

File: src/util/Util.js [hunk 2]
```diff
@@ -63,9 +63,22 @@ class Util {
    */
   static splitMessage(text, { maxLength = 2000, char = '\n', prepend = '', append = '' } = {}) {
     text = Util.verifyString(text, RangeError, 'MESSAGE_CONTENT_TYPE', false);
-    if (text.length <= maxLength) return [text];
-    const splitText = text.split(char);
-    if (splitText.some(chunk => chunk.length > maxLength)) throw new RangeError('SPLIT_MAX_LEN');
+    const hasElemLongerThanMaxLength = (arr, len) => arr.some(elem => elem.length > len);
```
[the remark this answers is not in the document]
Comment (u9g) [reply src/util/Util.js:66]: I mean array can be replaced with splitText too if I swap this line with the line above it.

[the remark this answers is not in the document]
Comment (SpaceEEC) [reply src/util/Util.js:66]: Maybe just get rid of that function completely then?
- Check the length of `text` before wrapping it in an array. (lines 67-68)
- Inline the function in line 70 and 81.

File: src/structures/interfaces/TextBasedChannel.js [hunk 3]
```diff
@@ -99,7 +99,8 @@ class TextBasedChannel {
    * Options for splitting a message.
    * @typedef {Object} SplitOptions
    * @property {number} [maxLength=2000] Maximum character length per message piece
-   * @property {string} [char='\n'] Character to split the message with
+   * @property {string | string[] | RegExp | RegExp[]} [char='\n'] Character(s) or Regex(s) to split the message with,
```
Comment (papaia) [inline src/structures/interfaces/TextBasedChannel.js:102]:
```suggestion
   * @property {string|string[]|RegExp|RegExp[]} [char='\n'] Character(s) or Regex(s) to split the message with,
```

Comment (kyranet) [reply src/structures/interfaces/TextBasedChannel.js:102]: Consistency with the rest of the library, mostly.

Comment (u9g) [reply src/structures/interfaces/TextBasedChannel.js]: sure, just curious, added

File: src/util/Util.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (u9g) [reply src/util/Util.js]: Done! added in cd094a5

File: src/util/Util.js [hunk 4]
```diff
@@ -63,9 +63,20 @@ class Util {
    */
   static splitMessage(text, { maxLength = 2000, char = '\n', prepend = '', append = '' } = {}) {
     text = Util.verifyString(text, RangeError, 'MESSAGE_CONTENT_TYPE', false);
-    if (text.length <= maxLength) return [text];
-    const splitText = text.split(char);
-    if (splitText.some(chunk => chunk.length > maxLength)) throw new RangeError('SPLIT_MAX_LEN');
+    if (text.length > maxLength) return [text];
```
Comment (SpaceEEC) [inline src/util/Util.js:66]:
```suggestion
    if (text.length > maxLength) return [text];
    let splitText = [text];
```

**Review by SpaceEEC — CHANGES_REQUESTED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**